buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/performance/PerformanceTestPlugin.kt

```diff
@@ -327,15 +314,34 @@ class PerformanceTestPlugin : Plugin<Project> {
             teamCityPassword = stringPropertyOrNull(PropertyNames.teamCityPassword)
         }
 
+        createAndWireCommitDistributionTasks(performanceTest)
+
         afterEvaluate {
-            result.configure {
+            performanceTest.configure {
                 branchName?.takeIf { it.isNotEmpty() }?.let { branchName ->
                     channel = channel + "-" + branchName
                 }
             }
         }
 
-        return result
+        return performanceTest
+    }
+
+    private
+    fun Project.createAndWireCommitDistributionTasks(performanceTest: TaskProvider<out PerformanceTest>) {
+        // The data flow here is:
+        // determineForkPointCommitBaseline.forkPointCommitBaseline -> performanceTest.forkPointCommitBaseline
+        // performanceTest.baselines -> buildCommitDistribution.commitBaseline
+        val determineForkPointCommitBaseline = tasks.register("${performanceTest.name}DetermineForkPointCommitBaseline", DetermineForkPointCommitBaseline::class)
+        val buildCommitDistribution = tasks.register("${performanceTest.name}BuildCommitDistribution", BuildCommitDistribution::class)
+        performanceTest.configure {
+            dependsOn(buildCommitDistribution)
+            forkPointCommitBaseline.set(determineForkPointCommitBaseline.flatMap { it.forkPointCommitBaseline })
+        }
+        buildCommitDistribution.configure {
+            dependsOn(determineForkPointCommitBaseline)
+            commitBaseline.set(performanceTest.flatMap { provider { it.baselines } })
```

Comment:
I would have expected this to be the determined baseline property from `determineForkPointCommitBaseline`.